blobfuse/src/BlobfuseGcCache.cpp

```diff
@@ -19,6 +19,18 @@ bool gc_cache::check_disk_space()
     double available;
     unsigned long long used;
 
+    static bool user_detected = false;
+    static bool is_sudo_user = false;
+
+    if (!user_detected) {
+        // If effective uid = 0 then we are running as sudo
+        if (geteuid() == 0)
+            is_sudo_user = true;
```

Comment:
is it a bad idea to make a global variable for the is_sudo_user? instead of having a flag here to make sure sudo detection is done or not?

Reply:
These are static variables inside a function so their scope is not global rather limited to this function only.